flask/launcher/src/main/java/net/corda/flask/launcher/Launcher.java

```diff
@@ -0,0 +1,127 @@
+package net.corda.flask.launcher;
+
+import lombok.SneakyThrows;
+import net.corda.flask.common.Flask;
+import net.corda.flask.common.ManifestEscape;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class Launcher {
```

Comment:
Are we _expected_ to override this class, or can we use it "as is"? If we're expected to override then maybe declare it as `abstract`? Perhaps also declare `launch()` as `final` since I'm not sure that "package private" matters much when you create a fat jar.

What happens if the subclass of `Launcher` "just happens" also to declare a function like this?
```java
public static void main(String[] args) {
    throw new RuntimeException("Victory is mine!");
}
```

Reply:
In general, I'm not going to prevent people from intentionally putting a bullet in their heads, code should only prevent unintentional mistakes. That being said, declaring a Launcher class with a separate `main` method won't have any effect since static methods are statically dispatched (the child class `main` method won't override anything).

This class is meant to be overridden *only* if you want to do insert some custom logic before starting the child process (like parsing the `node.conf` and scraping additional JVM arguments from there)

Reply:
no, sorry... in that case the `main` method will be invoked instead of `net.corda.flask.launcher.Launcher#main`, because the main class in the manifest is the launcher class. I don't believe this is an issue because you simply shouldn't do it and I cannot see how you could do this by mistake. But it could be addressed keeping the main class hardcoded and adding a separate manifest attribute for the launcher class name

Reply:
there is actually a more serious issue: this class is also used as a Java agent to stop the child process when the parent dies, this means that this whole jar is added to the classpath of the child process, polluting it with all its classes. My idea was that this would have limited side effects since the child process would only find himself with a `net.corda.flask.launcher` Java package that it would simply ignore but I now realize that it would also add to the classpath all of the (embedded) dependencies of the launcher class. As an example, in Corda, we woould find ourselves with 2 separate (in theory identical, but who knows) versions of Typesafe Config, once from the application code and one form the launcher code. 
I will move the agent to a dedicated jar as this is surely undesirable and will likely be the cause of subtle, hard to troubleshoot, bugs.